File: atcoder.jp/contests/abc159/tasks/abc159_d/main.py
from collections import Counter

N = int(input())
A = list(map(int, input().split()))


def nC2(n: int) -> int:
    if n < 2:
        return 0
    return n * (n - 1) // 2


counter = Counter(A)
total = sum(map(nC2, counter.values()))

ans_list = [total - nC2(counter[a]) + nC2(counter[a] - 1) for a in A]
print(*ans_list, sep="\n")

File: atcoder.jp/contests/abc159/tasks/abc159_d/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];
const print = (...messages: any) => console.log(...messages);

const sum = (data: number[]) => data.reduce((a, b) => a + b);
const Counter = <T>(arr: T[]) => {
  const counter = new Map<T, number>();
  for (const v of arr) counter.set(v, (counter.get(v) || 0) + 1);
  return counter;
};

const N = +input();
const A = input().split(" ").map(Number);

const nC2 = (n: number) => (n < 2 ? 0 : (n * (n - 1)) / 2);

const counter = Counter(A);
const total = sum([...counter.values()].map(nC2));

const ansArray = A.map(
  (a) => total - nC2(counter.get(a)!) + nC2(counter.get(a)! - 1)
);

print(ansArray.join("\n"));
